src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -168,10 +168,16 @@
 
     private GenomeLocSortedSet genomeLocSortedSet;
 
+    // information to keep track of when validating a GVCF
+    private int currentEnd;
```

Comment:
instead of keeping the end and the contig, keep a currentInterval and use the available math functions to check if the next interval abuts the current.